windows/windowsservercore/Dockerfile

```
@@ -32,19 +32,24 @@ SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPref
     Remove-Item -Path C:\temp -Recurse | Out-Null
 
+RUN powershell -Command "$javaMajorVersion = $env:JAVA_VERSION.substring(0,2); if ($javaMajorVersion -eq '17') { $options = '--compress=2 --add-modules ALL-MODULE-PATH' } elseif ($javaMajorVersion -eq '21') { $options = '--compress=zip-6 --add-modules ALL-MODULE-PATH' } elseif ($javaMajorVersion -eq '25') { $options = '--compress=zip-6 --add-modules java.base,java.logging,java.xml,java.se' } else { throw \"Unsupported Java version $env:JAVA_VERSION\" }; jlink --strip-java-debug-attributes $options --no-man-pages --no-header-files --output C:\javaruntime"
+
 ## Agent image target
 FROM mcr.microsoft.com/windows/servercore:"${WINDOWS_VERSION_TAG}" AS agent
 
-ARG JAVA_HOME="C:\openjdk-17"
-ENV JAVA_HOME=${JAVA_HOME}
+ARG JAVA_VERSION=17.0.15+6
+# Extract major version from JAVA_VERSION and set JAVA_HOME accordingly
+RUN $javaMajorVersion = $env:JAVA_VERSION.substring(0,2); `
```

Hadolint (2.12.0)

[error] Parsing stopped here. Invalid use of parentheses?

(SC1088)